core/src/main/java/com/sequenceiq/distrox/v1/distrox/converter/DistroXV1RequestToStackV4RequestConverter.java

```diff
@@ -99,9 +100,31 @@ public StackV4Request convert(DistroXV1Request source) {
         request.setTelemetry(getTelemetryRequest(source, environment, sdxClusterResponse));
         request.setGatewayPort(source.getGatewayPort());
         request.setExternalDatabase(getIfNotNull(source.getExternalDatabase(), databaseRequestConverter::convert));
+        request.setEnableLoadBalancer(source.isEnableLoadBalancer());
+        checkMultipleGatewayNodes(source);
         return request;
     }
 
+    private void checkMultipleGatewayNodes(DistroXV1Request distroXV1Request) {
+        if (distroXV1Request.getInstanceGroups() == null || distroXV1Request.getInstanceGroups().isEmpty()) {
+            return;
+        }
+        distroXV1Request.getInstanceGroups().forEach(ig -> {
+            if (InstanceGroupType.GATEWAY == ig.getType()) {
+                if (distroXV1Request.isEnableLoadBalancer()) {
+                    if (ig.getNodeCount() < 1) {
+                        throw new BadRequestException("Instance group with GATEWAY type must contain at least 1 node!");
```

Comment:
how is loadbalancing possible with 1 node?

Reply:
@bergerdenes Cloud load-balancing solution is used in multiple contexts in Cloudbreak. It acts as a static endpoint. If you move CM to a different node and create the cluster with the same name, you still get the same endpoint. In that context allowing a single node GATEWAY with load-balancer is a useful feature. 

The other contexts are,

1. Semi-private networks where the load-balancer is hosted in public network/IP, and the VMs are hosted in private network/IP.
2. Loadbalancing with more than one node, to achieve resiliency in case of node failures. Like Hive endpoint.